src/hera/workflow.py

```diff
@@ -320,3 +327,40 @@ def get_parameter(self, name: str) -> Parameter:
         if self.parameters is None or next((p for p in self.parameters if p.name == name), None) is None:
             raise KeyError(f"`{name}` is not a valid workflow parameter")
         return Parameter(name, value=f"{{{{workflow.parameters.{name}}}}}")
+
+    def to_dict(self) -> dict:
+        """Returns the JSON/dictionary representation of the workflow"""
+        return self.build().to_dict()
+
+    def to_yaml(self) -> str:
+        """Returns a YAML representation of the workflow"""
+        dict_repr = self.to_dict()
+        new_dict_repr: dict = {}
+        _traverse(dict_repr, new_dict_repr, _camel_case)
```

Comment:
There is actually a function in the Argo sdk that already contains this logic. See https://github.com/argoproj/argo-workflows/blob/6b086368f6480a2de5e2d43eec73514de0ad01ac/sdks/python/client/argo_workflows/model_utils.py#L1648

when serialize it set to true, it correctly uses the model attribute map to convert python attributes to json/yaml ones.